raiden/raiden_protocol.py

```diff
@@ -21,15 +23,72 @@ class RaidenProtocol(object):
     try_interval = 1.
     max_tries = 5
     max_message_size = 1200
+    short_delay = .1
 
     def __init__(self, transport, discovery, raiden):
         self.transport = transport
         self.discovery = discovery
         self.raiden = raiden
+        self.queued_messages = Queue()
+        self.stop_event = AsyncResult()
 
         self.number_of_tries = dict()  # msg hash: count_tries
         self.sent_acks = dict()  # msghash: Ack
 
+        gevent.spawn(self._send_queued_messages)
+
+    def _send_queued_messages(self):
+        countdown_to_send = self.try_interval / self.short_delay
+        countdown = 0
+
+        stop = None
+        while stop is None:
+            # blocks waiting for data in queue, don't remove data from
+            # the queue just now because sending can still fail.
+            receiver_address, message = self.queued_messages.peek()
```

Comment:
`Queue.peek()` blocks until a new element is added into the queue, the `stop_event` will be ineffective here.

Here is a sketch for the change:
```python
while True:
    ready = gevent.wait([self.stop_event, self.queued_messages])  # this wont work, need to wrap the queue

    if self.stop_event in ready:
        return

    receiver_address, message = self.queued_messages.peek()
```

The `Queue` object does not implement the wait protocol, we will probably need a proxy of some sort to make it work.